### ISSUE TITLE
Integrate sprig template function library

### ISSUE THREAD

**Issue by hamiltont** [fstab/grok_exporter#79]
The templates directory is nice, but given the range of data that could pass through grok_exporter perhaps something like http://masterminds.github.io/sprig/ should be incorporated. For example, I'm trying to use the output of `subtract` to slice, but unfortunately it outputs float64 while slice requires an int type. Having a larger body of helper functions would allow handling more edge case situations

**Comment by fstab** [comment]
I agree, more functions would be good. I'll have a look if it's possible to implement the Sprig functions in grok_exporter.
